src/test/java/com/wikia/webdriver/testcases/blogtests/BlogTests.java

```diff
@@ -1,123 +1,120 @@
-    base.loginAs(credentials.userName4, credentials.password4, wikiURL);
-    String blogTitle = PageContent.BLOG_POST_NAME_PREFIX + base.getTimeStamp();
-    String blogContent = PageContent.BLOG_CONTENT + base.getTimeStamp();
-    UserProfilePageObject userProfile = base.openProfilePage(credentials.userName4, wikiURL);
+    UserProfilePage userProfile = new UserProfilePage().open(User.BLOGS.getUserName());
     userProfile.clickOnBlogTab();
     SpecialCreatePage createBlogPage = userProfile.clickOnCreateBlogPost();
     VisualEditModePageObject visualEditMode = createBlogPage.populateTitleField(blogTitle);
     visualEditMode.addContent(blogContent);
-    BlogPageObject blogPage = visualEditMode.submitBlog();
+    BlogPage blogPage = visualEditMode.submitBlog();
     blogPage.getBlogTitle();
     blogPage.verifyContent(blogContent);
   }
 
-  @Test(
-      dataProviderClass = ArticleDataProvider.class,
-      dataProvider = "articleTitles",
-      groups = {"BlogTests_002", "BlogTests"})
-  @Execute(asUser = User.USER)
-  public void BlogTests_002_addByUrl(String blogTitle) {
-    WikiBasePageObject base = new WikiBasePageObject();
-    String blogContent = PageContent.BLOG_CONTENT + base.getTimeStamp();
-    String randomBlogTitle = blogTitle + base.getTimeStamp();
-    SpecialCreatePage createBlogPage = base.openSpecialCreateBlogPage(wikiURL);
+  @Test(dataProviderClass = ArticleDataProvider.class, dataProvider = "articleTitles")
+  @Execute(asUser = User.BLOGS)
+  public void UserCanCreateBlogsWithSpecialCharacters (String blogTitle) {
+    String blogContent = PageContent.BLOG_CONTENT + DateTime.now().getMillis();
+    String randomBlogTitle = blogTitle + DateTime.now().getMillis();
+    SpecialCreatePage createBlogPage = new SpecialCreatePage().open();
     VisualEditModePageObject visualEditMode = createBlogPage.populateTitleField(randomBlogTitle);
     visualEditMode.addContent(blogContent);
-    BlogPageObject blogPage = visualEditMode.submitBlog();
+    BlogPage blogPage = visualEditMode.submitBlog();
     blogPage.getBlogTitle();
     blogPage.verifyContent(blogContent);
   }
 
-  @Test(groups = {"BlogTests_003", "BlogTests"})
-  @Execute(asUser = User.USER)
-  public void BlogTests_003_editFromProfile() {
-    WikiBasePageObject base = new WikiBasePageObject();
-    String blogContent = PageContent.BLOG_CONTENT + base.getTimeStamp();
-    UserProfilePageObject userProfile = base.openProfilePage(credentials.userName, wikiURL);
-    userProfile.clickOnBlogTab();
-    BlogPageObject blogPage = userProfile.openFirstPost();
+  @Execute(asUser = User.BLOGS)
+  public void UserCanEditBlogPost() {
+    String blogTitle = PageContent.BLOG_POST_NAME_PREFIX + DateTime.now().getMillis();
+    String blogContent = PageContent.BLOG_CONTENT + DateTime.now().getMillis();
+    new ArticleContent(User.BLOGS).push(blogContent,
+        String.format("User_blog:%s/%s", User.BLOGS.getUserName(), blogTitle));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "User_blog:%s/%s" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)